src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -35,74 +38,125 @@
         requireNonNull(editPersonDescriptor);
+        requireNonNull(personType);
 
         this.index = index;
-        this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+        this.editPersonDescriptor = editPersonDescriptor;
+        this.personType = personType;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        // Note: getFilteredPersonList() no longer exists.
-        // TODO: Replace with getFilteredStudentList() or getFilteredTutorList() depending on which list you want.
-        List<Person> lastShownList = model.getFilteredPersonList();
+        if (personType.equals(PersonType.TUTOR)) {
+            List<Tutor> lastShownList = model.getFilteredTutorList();
+
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_TUTOR_DISPLAYED_INDEX);
+            }
+
+            Tutor tutorToEdit = lastShownList.get(index.getZeroBased());
+            Tutor editedTutor = createEditedTutor(tutorToEdit, (EditTutorDescriptor) editPersonDescriptor);
 
-        if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            if (!tutorToEdit.isSamePerson(editedTutor) && model.hasTutor(editedTutor)) {
+                throw new CommandException(MESSAGE_DUPLICATE_TUTOR);
+            }
+
+            model.setTutor(tutorToEdit, editedTutor);
+            model.updateFilteredTutorList(PREDICATE_SHOW_ALL_TUTORS);
+
+            return new CommandResult(String.format(MESSAGE_EDIT_TUTOR_SUCCESS, editedTutor));
         }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+        if (personType.equals(PersonType.STUDENT)) {
+            List<Student> lastShownList = model.getFilteredStudentList();
+
+            if (index.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
+            }
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+            Student studentToEdit = lastShownList.get(index.getZeroBased());
+            Student editedStudent = createEditedStudent(studentToEdit, (EditStudentDescriptor) editPersonDescriptor);
+
+            if (!studentToEdit.isSamePerson(editedStudent) && model.hasStudent(editedStudent)) {
+                throw new CommandException(MESSAGE_DUPLICATE_STUDENT);
+            }
+
+            model.setStudent(studentToEdit, editedStudent);
+            model.updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
+
+            return new CommandResult(String.format(MESSAGE_EDIT_STUDENT_SUCCESS, editedStudent));
```

Comment:
Perhaps you could use a switch case statement for this as there won't be issues with identifying the enum